File: Code coach problems/Hard/C++/Security.cpp
#include <iostream>
#include <string>
using namespace std;
int main()
{
	string s;
	cin >> s;
	int i = 0;
	for(; i < s.size(); i++)
	{
		if(s[i] == 'T')
		{
			int p = i;
			while(i < s.size() && s[i] != '$' && s[i] != 'G')
				i++;
			if(i != s.size() && s[i] == '$')
				break;
				
			i = p;
		}
		else if(s[i] == '$')
		{
			int p = i;
			
			while(i < s.size() && s[i] != 'T' && s[i] != 'G')
				i++;
				
			if(i != s.size() && s[i] == 'T')
				break;
			i = p;
		}
	}
	
	if(i == s.size())
		cout << "quiet";
	else 
		cout << "ALARM";
}

File: Code coach problems/Hard/Python/Security.py
s = input()
t = 1
for i in range (len(s)):
	if s[i] == 'T':
		p = i
		while p < len(s) and s[p] != '$' and s[p] != 'G':
			p+=1
		if p != len(s) and s[p] == '$':
			t = 0
			break
	elif s[i] == "$":
		p = i
		while p < len(s) and s[p] != 'T' and s[p] != 'G':
			p+=1
		if p != len(s) and s[p] == 'T':
			t = 0
			break

if t == 1:
	print("quiet")
else:
	print("ALARM")
